src/Google.Cloud.Functions.Invoker/EntryPoint.cs

```diff
@@ -60,19 +61,77 @@ public static async Task<int> StartAsync(Assembly functionAssembly, string[] arg
+        /// <typeparam name="TFunction">The function type to host.</typeparam>
+        public static IHostBuilder CreateHostBuilder<TFunction>() => CreateHostBuilder(typeof(TFunction));
+
+        /// <summary>
+        /// Creates an <see cref="IWebHostBuilder"/> for the function type specified by
+        /// <paramref name="functionType"/>. This is a convenience method for calling
+        /// <see cref="CreateHostBuilder(IReadOnlyDictionary{string, string}, Assembly)"/>, passing in a dictionary
+        /// that only contains the function target environment variable and the assembly
+        /// containing the function type. This method is primarily used for writing integration tests
+        /// where the function type isn't known at compile time.
+        /// </summary>
+        /// <param name="functionType">The function type to host.</param>
+        /// <returns>A host builder that can be used for integration testing.</returns>
+        public static IHostBuilder CreateHostBuilder(Type functionType)
+        {
+            Preconditions.CheckNotNull(functionType, nameof(functionType));
+            var functionTarget = functionType.FullName ?? throw new ArgumentException("Target function type has no name.");
+            var environment = new Dictionary<string, string> { { FunctionTargetEnvironmentVariable, functionTarget } };
+            return CreateHostBuilder(environment, functionType.Assembly);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="IWebHostBuilder"/> as if the environment variables are set as per <paramref name="environment"/>.
+        /// The actual system environment variables are ignored. Command line arguments are not supported, as (by design) everything
+        /// that can be specified on the command line can also be specified via environment variables.
+        /// This method is primarily used for writing integration tests where a high degree of control is required,
+        /// for example to simulate the difference between running in a container or not.
+        /// </summary>
+        /// <param name="environment">The fake environment variables to use when constructing the host builder.</param>
+        /// <param name="functionAssembly">The assembly containing the target function. May be null, in which case the calling assembly
+        /// is used.</param>
+        /// <returns>A host builder that can be used for integration testing.</returns>
+        public static IHostBuilder CreateHostBuilder(IReadOnlyDictionary<string, string> environment, Assembly? functionAssembly = null)
+        {
+            var variables = ConfigurationVariableProvider.FromDictionary(environment);
+            var functionEnvironment = FunctionEnvironment.Create(functionAssembly ?? Assembly.GetCallingAssembly(), new string[0], variables);
+            return CreateHostBuilder(functionEnvironment);
+        }
+
+        // TODO: Move to FunctionEnvironment?
```

Comment:
It's almost like we need the builder for the builder :). Yes, I think It'll be better in FunctionEnvironment.